File: Introductory Problems/permutations.cpp
// code by savir singh

#include <bits/stdc++.h>
using namespace std;

int main()
{
    long long n;
    cin >> n;

    if (n==1) {
        cout << 1;
    }
    else if (n<4) {
        cout << "NO SOLUTION";
    }
    else {
        vector<int> even;
        vector<int> odd;
        for (int i=n; i>0; i--) {
            if (i%2==0) {
                even.push_back(i);
            }
            else {
                odd.push_back(i);
            }
        }
        for (int num:odd) {
            cout << num << " ";
        }
        for (int num:even) {
            cout << num << " ";
        }
    }
}

File: Introductory Problems/permutations.py
# code by savir singh

import sys
n = int(sys.stdin.readline())

if n == 1:
    print(1)
    sys.exit()
if n <= 3:
    print("NO SOLUTION")
    sys.exit()
else:
    even = []
    odd = []
    for i in range(n, 0, -1):
        if i % 2 == 0:
            even.append(i)
        else:
            odd.append(i)

    print(str(odd + even).replace('[', '').replace(']', '').replace(',', ''))
